app/scripts/lib/segment/analytics.js

```diff
@@ -0,0 +1,219 @@
+import removeSlash from 'remove-trailing-slash';
+import looselyValidate from '@segment/loosely-validate-event';
+import { isString } from 'lodash';
+
+// eslint-disable-next-line
+const setImmediate = global.setImmediate || process.nextTick.bind(process);
+const noop = () => ({});
+const makeid = () => {
+  let result = '';
+  const characters =
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < 20; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+};
+
+export default class Analytics {
```

Comment:
I think this comment is a bit late but might have been nice to add the original file first, and then a new commit with changes. 

do you think it would be helpful if we added a comment to explain where the code originated from and include which version it was?

Reply:
Agreed. At least for reviewing this would be very helpful

Reply:
I ended up using a random diff checker I found online 

code changes look good to me. Aside from the axios <> fetch swap itself, some things I observed were:  
- `screen` and `alias` methods were removed 
- we are no longer returning `setImmediate(callback);` in `enqueue` if `!this.enable`. This looks like a good clean up to me 
- no more retry logic. Q: why was this removed? @jpuri

Reply:
I wish I saw this comment earlier, I had to open the code side by side. But yeah, code cleanup looks good.

Reply:
Methods that we do not use in metamask are removed, the file is small subset of what we use.
retry depends on axiosRetry thus it is also removed with axios library.

Reply:
After discussion with @PeterYinusa , I added retry code to the PR.